src/test/resources/log4j2-test.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
This probably needs to be a separate discussion but do we want to use log4j for test logging?  I would think if we're going to provide runtime logging for the extensions, we'd want to use a facade that didn't require an implementation.  For test logging should we  align to that?

Reply:
I assume Marc used log4j 2 simply because that's what we do in JUnit 5, and I would have done the same.

Pioneer should use JUL (Java Util Logging) just like JUnit 5 does. Otherwise, Pioneer will require a dependency on a logging framework, and I think we should avoid that.

Note that we're only using log4j as the `LogManager` for JUL within our own test suite: that has no impact on users of Pioneer.

Reply:
I agree with @smoyer64: we should introduce a `jupiter` subpackage for things specific to JUnit Jupiter.

Reply:
Would Google [Flogger](https://github.com/google/flogger) be worth considering? It has [many advantages](https://google.github.io/flogger/benefits) over other logging APIs, but I'm unsure if it's worth the extra dependency.

Reply:
Ummm... you're missing the point. 😉 

Flogger wouldn't help: we don't want to use any logging _API_ other than JUL.

Rationale: we don't want JUnit or Pioneer to force users to add additional dependencies to their projects. JUL is always present in the JDK, and logging frameworks such as log4j provide "logger" _implementations_ for JUL.

Reply:
> Ummm... you're missing the point. 😉

Flogger wouldn't help: we don't want to use any logging API other than JUL.

Rationale: we don't want JUnit or Pioneer to force users to add additional dependencies to their projects. JUL is always present in the JDK, and logging frameworks such as log4j provide "logger" implementations for JUL.

Oops, didn't quite realise that! Thanks for the clarification @sbrannen. 😅

Reply:
> I assume Marc used log4j 2 simply because that's what we do in JUnit 5, and I would have done the same.

I just want to make sure we're all in agreement with that assumption - I don't have an issue with it.

> Pioneer should use JUL (Java Util Logging) just like JUnit 5 does. Otherwise, Pioneer will require a dependency on a logging framework, and I think we should avoid that.

Pioneer has the added caveat that we can use external dependencies that are "provided" scoped, so we could use another facade.  (I'm not suggesting this is one of those times and I don't know that Flogger would be my choice if we did).  I think during our telecon we all agreed that this was to be used very sparingly.

Reply:
Right: JUnit and Pioneer should not log extensively.

And although the JUL _API_ isn't the nicest (maybe the worst?), that's really just a burden on the developers of JUnit and Pioneer. Users of the JUnit and Pioneer are then free to use whatever logging framework they like (and provide a bridge from JUL to their logging backend).

Reply:
> Pioneer has the added caveat that we can use external dependencies that are "provided" scoped, so we could use another facade.

True. But I don't think logging would fall into that category of dependencies. As I recall, we agreed that Pioneer could have _optional_ dependencies on other third-party frameworks (e.g., database connectivity, server APIs, etc.).

Of course, if an optional third-party dependency brings in logging dependencies transitively, that would be a different story. 😉

Reply:
Yes ... "optional" (I'd use "provided" scope in a Maven project).  Won't Gradle scope the transitive dependencies for an "optional" dependency as "optional" as well?

Reply:
Gradle doesn't actually have an `optional` scope. That's more like `compileOnly` & Co.

But there's a plugin from the Spring team that _implements_ an `optional` scope in Gradle that gets exported as Maven's `optional` scope.

And... I _think_ an upcoming version of Gradle might support `optional` out of the box, though I imagine @marcphilipp can better answer that. 😉